src/compiler/factory.ts

```diff
@@ -0,0 +1,125 @@
+/// <reference path="core.ts"/>
+/// <reference path="utilities.ts"/>
+
+/* @internal */
+namespace ts {
+    let NodeConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
+    let SourceFileConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;

```

Comment:
Is there any reason you need to do this again instead of leveraging `ts.createNode`?